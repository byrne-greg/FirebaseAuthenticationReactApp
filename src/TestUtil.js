import React from "react";
import { MemoryRouter } from "react-router-dom";

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter };
