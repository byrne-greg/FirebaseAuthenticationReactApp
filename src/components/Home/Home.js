import React from "react";
import { withRouter } from "react-router-dom";
import { useAuthorization } from "../Session";
// import * as ROUTES from "../../constants/routes";

const HomePage = ({ history }) => {
  const authStrategy = authUser => !!authUser;
  const isAuthorized = useAuthorization(authStrategy);
  return isAuthorized ? (
    <div>
      <h1>Home Page</h1>
      <p>The Home Page is accessible by every signed in user.</p>
    </div>
  ) : null;
  // by pushing here, the useAuthorization may be still be in a call and the component has unmounted,
  // leaving an error in the console
  // <>{history.push(ROUTES.SIGN_IN)}</>
};

export default withRouter(HomePage);
