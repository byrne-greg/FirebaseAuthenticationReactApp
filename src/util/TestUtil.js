import React from "react";
import { MemoryRouter } from "react-router-dom";
import { FirebaseContext } from "../components/Firebase";
import { AuthUserContext } from "../components/Session";

const withMockFirebase = (mockFirebase, Component) => (
  <FirebaseContext.Provider value={mockFirebase}>
    {Component}
  </FirebaseContext.Provider>
);

const withMockAuthUserContext = (value, Component) => (
  <AuthUserContext.Provider value={value}>{Component}</AuthUserContext.Provider>
);

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter, withMockFirebase, withMockAuthUserContext };
