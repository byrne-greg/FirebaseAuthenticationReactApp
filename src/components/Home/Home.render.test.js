import React from "react";
import { render, cleanup } from "@testing-library/react";
import { withTestRouter, withMockAuthUserContext } from "../../util/TestUtil";
import Home from ".";

afterEach(cleanup);

describe("Home render tests", () => {
  it("when user is unauthenticated, the unauthorized page is rendered", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<Home />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  it("when user is authenticated, the home page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(
      withTestRouter(withMockAuthUserContext({ mockUser: true }, <Home />))
    );

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });
});
