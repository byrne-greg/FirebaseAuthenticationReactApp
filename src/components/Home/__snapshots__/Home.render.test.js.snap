// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render tests when user is authenticated, the home page renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Home Page
      </h1>
      <p>
        The Home Page is accessible by every signed in user.
      </p>
    </div>
  </div>
</body>
`;

exports[`Home render tests when user is unauthenticated, the unauthorized page is rendered 1`] = `
<body>
  <div>
    <div>
      <h1>
        You are not allowed to access this page
      </h1>
    </div>
  </div>
</body>
`;

exports[`Navigation render tests when user is authenticated, the authenticated user navigation bar renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Home Page
      </h1>
      <p>
        The Home Page is accessible by every signed in user.
      </p>
    </div>
  </div>
</body>
`;

exports[`Navigation render tests when user is unauthenticated, the public navigation bar renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        You are not allowed to access this page
      </h1>
    </div>
  </div>
</body>
`;
